packages/api/tests/endpoints/providers/update-provider.js

```diff
@@ -60,14 +80,24 @@ test.beforeEach(async (t) => {
     ...fakeProviderFactory(),
     cmKeyId: 'key',
   };
-  await providerModel.create(t.context.testProvider);
+  await request(app)
```

Comment:
The alternative is to re-implement a  version of that addition code (e.g. manual knex write with translation from the test object + dynamo write) for this and deletes.   

I agree the post portion of this doesn't feel great, but *really* don't want to un-DRY this into this and the deletes tests.   Thoughts on abstracting at a minimum this portion: 

```
    const createObject = await postgresProviderFromCumulusProvider(data);
    await knex.transaction(async (trx) => {
      await trx(tableNames.providers).insert(createObject);
      record = await providerModel.create(data);
    });
``` 

from endpoints/provider into a method that can be used?

Reply:
I could be wrong, but that doesn't quite seem necessary.

For a test setup, do we really need to use the transactional write? If the writes don't apply consistently, then we'd get test failures. So could we just do this instead?

```
   const createObject = await postgresProviderFromCumulusProvider(data);
   await knex.insert(tableNames.providers).insert(createObject);
   await providerModel.create(data);
```

Reply:
> For a test setup, do we really need to use the transactional write? 

🤔   .... no, we don't need to do so.  

>  So could we just do this instead?

Yes, but.   On principle, the base logic in that creation routine ( the transaction is beside the point) builds an assumption into the test that still doesn't feel quite right -- if that logic is changed in the endpoint for some reason have to remember to update the  or the units break (perhaps subtly).   

I think really the (my?) issue here is with the way these tests are setup, generally (and we have a ticket to resolve that), and when that's fixed this will be less of a concern. 

I'll conceded the point with mild duress here as I don't think it's legit going to cause us real grief, and the practical difference isn't going to drive user value.

Reply:
Yeah, it's not the hill I'm going to die on. It just feels weird to me to be subtly testing a separate endpoint in the setup for these tests.

Reply:
b0f96bbac